server/src/main/java/io/unitycatalog/server/service/ModelService.java

```diff
@@ -3,102 +3,271 @@
+
+import io.unitycatalog.server.utils.IdentityUtils;
+import lombok.SneakyThrows;
+
+import static io.unitycatalog.server.model.SecurableType.CATALOG;
+import static io.unitycatalog.server.model.SecurableType.METASTORE;
+import static io.unitycatalog.server.model.SecurableType.REGISTERED_MODEL;
+import static io.unitycatalog.server.model.SecurableType.SCHEMA;
 
 @ExceptionHandler(GlobalExceptionHandler.class)
 public class ModelService {
 
   private static final ModelRepository MODEL_REPOSITORY = ModelRepository.getInstance();
 
-  public ModelService() {}
+  private static final SchemaRepository SCHEMA_REPOSITORY = SchemaRepository.getInstance();
+  private static final CatalogRepository CATALOG_REPOSITORY = CatalogRepository.getInstance();
+
+  private final UnityCatalogAuthorizer authorizer;
+  private final UnityAccessEvaluator evaluator;
+
+  @SneakyThrows
+  public ModelService(UnityCatalogAuthorizer authorizer) {
+    this.authorizer = authorizer;
+    evaluator = new UnityAccessEvaluator(authorizer);
+  }
 
   @Post("")
-  public HttpResponse createRegisteredModel(CreateRegisteredModel createRegisteredModel) {
+  @AuthorizeExpression("""
+          #authorize(#principal, #catalog, OWNER) ||
+          (#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
+          #authorize(#principal, #catalog, USE_CATALOG) && #authorizeAll(#principal, #schema, USE_SCHEMA, CREATE_MODEL)
+          """)
+  public HttpResponse createRegisteredModel(
+          @AuthorizeKeys({
+                  @AuthorizeKey(value = SCHEMA, key = "schema_name"),
+                  @AuthorizeKey(value = CATALOG, key = "catalog_name")
+          })
+          CreateRegisteredModel createRegisteredModel) {
     assert createRegisteredModel != null;
     RegisteredModelInfo createRegisteredModelResponse =
         MODEL_REPOSITORY.createRegisteredModel(createRegisteredModel);
+    initializeAuthorizations(createRegisteredModelResponse);
     return HttpResponse.ofJson(createRegisteredModelResponse);
   }
 
   @Get("")
+  @AuthorizeExpression("#defer")
   public HttpResponse listRegisteredModels(
       @Param("catalog_name") Optional<String> catalogName,
       @Param("schema_name") Optional<String> schemaName,
       @Param("max_results") Optional<Integer> maxResults,
       @Param("page_token") Optional<String> pageToken) {
-    return HttpResponse.ofJson(
-        MODEL_REPOSITORY.listRegisteredModels(catalogName, schemaName, maxResults, pageToken));
+
+    ListRegisteredModelsResponse listRegisteredModelsResponse =
+            MODEL_REPOSITORY.listRegisteredModels(catalogName, schemaName, maxResults, pageToken);
+    filterModels("""
+            #authorize(#principal, #metastore, OWNER) ||
+            #authorize(#principal, #catalog, OWNER) ||
+            (#authorize(#principal, #catalog, USE_CATALOG) && #authorize(#principal, #schema, OWNER)) ||
+            #authorize(#principal, #registered_model, OWNER) ||
```

Comment:
Registered model owner needs USE_CATALOG and USE_SCHEMA privileges. It may be simpler to combine with the next rule by making `#authorize(#principal, #registered_model, EXECUTE)` to `#authorize(#principal, #registered_model, OWNER, EXECUTE)`